srcpkgs/cutechess-gui/template

```diff
@@ -0,0 +1,25 @@
+# Template file for 'cutechess-gui'
+pkgname=cutechess-gui
+version=1.0.0
+revision=1
+build_style=qmake
+makedepends="qt5-devel qt5-svg-devel"
```

Comment:
`qt5-qmake` seems to be missing for cross-compilation

Reply:
Not really sure what I am to do about this

Reply:
just add `qt5-qmake` to `hostmakedepends`. That way, it will always be installed for the host-architecture.

Reply:
Okay so while adding `qt5-qmake` to `hostmakedepends` means there is now qmake, compilation for those archs still fail due to what looks like inability to find other dependencies in `makedepends`.

Can I get a brief understanding of why cross-compilation can't make use of the information that's in the template? Why is it that many packages are also doing:

```sh
if [ "$CROSS_BUILD" ]; then
        hostmakedepends+=" qt5-host-tools qt5-devel"
fi
```

when `qt5-devel` is already in `makedepends`? And do I also need to include `qt5-host-tools`? Sorry my understanding of qt is also lacking, and I don't want to hog travis.

Reply:
I don't know if qt5-host-tools is needed for this package but it is needed for many so the probability is high. If `xbps-src` is not cross-compiling, alle packages (`host-` and `makedepends`) are installed into the same environment. However when it is cross-compiling, there's a separate cross-compilation environment which contains tools and libraries for the target architecture. So then everything is split in two, `hostmakedepends` go into the host environment, `makedepends` into the cross-environment. If tools that are to be executed in the host environment are in `makedepends`, they will not be available in the host environment when cross-compiling so they have to be explicitely added to `hostmakedepends`.

Reply:
Thanks, that makes sense. The `qt5-host-tools` seems necessary, but the rest I still find weird though because `-devel` packages aren't supposed to contain tools?

Reply:
qt5-devel in makedepends *should* be enough but qmake (is broken and) uses host headers when cross-compiling. There is other, more *right* way to address that than adding to hostmakedepends, but [not comfortable](https://github.com/void-linux/void-packages/blob/master/srcpkgs/liteide/template) either.